controllers/ibmpowervscluster_controller_test.go

```diff
@@ -433,6 +441,341 @@ func TestIBMPowerVSClusterReconciler_delete(t *testing.T) {
+				clusterScope.IBMVPCClient = mockVPC
+				return clusterScope
+			},
+			reconcileResult: reconcileResult{
+				error: fmt.Errorf("loadbalancer not found"),
+			},
+
+			conditions: capiv1beta1.Conditions{
+				capiv1beta1.Condition{
+					Type:               infrav1beta2.LoadBalancerReadyCondition,
+					Status:             "False",
+					Severity:           capiv1beta1.ConditionSeverityError,
+					LastTransitionTime: metav1.Time{},
+					Reason:             infrav1beta2.LoadBalancerReconciliationFailedReason,
+					Message:            "loadbalancer not found",
+				},
+				getVPCReadyCondition(),
+				getVPCSGReadyCondition(),
+				getVPCSubnetReadyCondition(),
+			},
+		},
+		{
+			name: "when Reconciling LoadBalancer and loadbalancer is ready",
+			powerVSClusterScopeFunc: func() *scope.PowerVSClusterScope {
+				clusterScope := &scope.PowerVSClusterScope{
+					IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+						Spec: infrav1beta2.IBMPowerVSClusterSpec{
+							VPC: &infrav1beta2.VPCResourceReference{
+								Region: ptr.To("us-south"),
+							},
+							VPCSubnets: []infrav1beta2.Subnet{
+								{
+									ID: ptr.To("subnet-id"),
+								},
+							},
+							LoadBalancers: []infrav1beta2.VPCLoadBalancerSpec{
+								{
+									ID: ptr.To("lb-id"),
+								},
+							},
+						},
+						Status: infrav1beta2.IBMPowerVSClusterStatus{
+							VPC: &infrav1beta2.ResourceReference{
+								ID: ptr.To("vpcID"),
+							},
+						},
+					},
+				}
+				mockVPC := mock.NewMockVpc(gomock.NewController(t))
+				mockVPC.EXPECT().GetVPC(gomock.Any()).Return(&vpcv1.VPC{Status: ptr.To("active")}, nil, nil)
+				mockVPC.EXPECT().GetSubnet(gomock.Any()).Return(&vpcv1.Subnet{Name: ptr.To("subnet1"), Status: ptr.To("active")}, nil, nil)
+				mockVPC.EXPECT().GetLoadBalancer(gomock.Any()).Return(&vpcv1.LoadBalancer{
+					ID:                 ptr.To("lb-id"),
+					Name:               ptr.To("lb"),
+					ProvisioningStatus: ptr.To("active"),
+				}, nil, nil)
+				clusterScope.IBMVPCClient = mockVPC
+				return clusterScope
+			},
+			conditions: capiv1beta1.Conditions{
+				getVPCLBReadyCondition(),
+				getVPCReadyCondition(),
+				getVPCSGReadyCondition(),
+				getVPCSubnetReadyCondition(),
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			reconciler := &IBMPowerVSClusterReconciler{
+				Client: testEnv.Client,
+			}
+			clusterScope := tc.powerVSClusterScopeFunc()
+			ch := make(chan reconcileResult, 1)
+			pvsCluster := &powerVSCluster{
+				cluster: clusterScope.IBMPowerVSCluster,
+			}
+			wg := &sync.WaitGroup{}
+			wg.Add(1)
+			reconciler.reconcileVPCResources(clusterScope, pvsCluster, ch, wg)
+			wg.Wait()
+			close(ch)
+			g.Expect(<-ch).To(Equal(tc.reconcileResult))
+			if len(tc.conditions) > 0 {
+				ignoreLastTransitionTime := cmp.Transformer("", func(metav1.Time) metav1.Time {
```

Comment:
For my understanding, why are we adding this condition?

Reply:
If you mean why cmp.Transfer() it simply ignores the LastTransition time as we cannot compare two time.Now() values.

Reply:
I'm trying to understand why `g.Expect(pvsCluster.cluster.GetConditions()).To(Equal(tc.conditions))` not work?

Reply:
Yes, It works, I will update.